### ISSUE TITLE
about fp loss

### ISSUE THREAD

**Issue by mooncakehe** [ElementAI/LCFCN#7]
I have a question about fp loss.
I kown you pick those blobs which do not contain points(marked label) as false positive dataset.
But if a blob do not contail a point while it truly in goals. maybe besides the point. put this blob as fp is not accurately true.
do you have some suggestion?
thanks!

**Comment by mooncakehe** [comment]
I read the paper again. the blob will be remove just retain the blob contains point .
the probs will drop down and can not form a blob. pretty good!

**Comment by mooncakehe** [comment]
# split_mode loss
    if blob_dict["n_multi"] > 0:
        loss += compute_split_loss(S_log, S, points, blob_dict)

    # Global loss 
    S_npy = ut.t2n(S.squeeze())
    points_npy = ut.t2n(points).squeeze()
    for l in range(1, S.shape[1]):
        points_class = (points_npy==l).astype(int)

        if points_class.sum() == 0:
            continue

        T = watersplit(S_npy[l], points_class)
        # imsave(batch["name"][0], T*255)
        T = 1 - T
        # print(batch["image_path"][0])
        scale = float(counts.sum())
        loss += float(scale) * F.nll_loss(S_log, torch.LongTensor(T).cuda()[None],
                        ignore_index=1, reduction='elementwise_mean')

what is the difference between the split loss and globa loss?
how the scale works? 
it will narrow the blob?
please tell me!

**Comment by IssamLaradji** [comment]
Hi there!

> 1.  what is the difference between the split loss and global loss?

The global loss is just another variation of the split loss. There is a local split loss and a global split loss.

- The **local split loss** makes the model set the split boundaries **that only separate between objects that are in the same blob** as background.
- The **global split loss** makes the model set the split boundaries **that separate all objects in the image** as background.

> 2. how the scale works?

The model automatically learns the scale. It will find the right scale that allows it to output one blob per object.

> 3. it will narrow the blob?

Yes, without the split loss the model will output one big blob containing many objects. With the split loss, the model will output smaller blobs to make sure there is one blob per object.

**Comment by mooncakehe** [comment]
yeah I figure out the the split loss.
the scale in local split loss is the num of pixs in blob. can I deem the bigger the blob is .the easier the blob will be separate? and the scale in global split loss is the num of the points.
my thoughts may not be mature enough. whether there is a problem between the ratio between positive samples and negative samples? the num of negative samples is bigger than positive samples.is helpful if use focal loss?
thanks for your detailed and fast explaination.

**Comment by IssamLaradji** [comment]
Hiya,
1. Yes, I think that the bigger the blob the easier for it to be separated, as networks often struggle with small resolutions.
2. The scale in the global split loss is indeed the number of points.
3. You are right that there is imbalance between positive and negative samples. It would be interesting to optimize the focal loss to handle that.
